doc/modules/core.rst

```diff
@@ -166,10 +166,153 @@ Internally, any sorting object can construct 2 internal caches:
 SortingAnalyzer
 ---------------
 
-The :py:class:`~spikeinterface.core.SortingAnalyzer` is the class which connects a :code:`Recording` and a :code:`Sorting`.
+The :py:class:`~spikeinterface.core.SortingAnalyzer` class is the core object to combine a
+:py:class:`~spikeinterface.core.BaseRecording` and a :py:class:`~spikeinterface.core.BaseSorting` object.
+This is the first step for additional analyses, and the basis of several postprocessing and quality metrics
+computations.
+
+The :py:class:`~spikeinterface.core.SortingAnalyzer` provides a convenient API to access the underlying
+:py:class:`~spikeinterface.core.BaseSorting` and :py:class:`~spikeinterface.core.BaseRecording` information,
+and it supports several extensions (derived from the :py:class:`~spikeinterface.core.AnalyzerExtension` class)
+to perform further analysis.
+
+Importantly, the :py:class:`~spikeinterface.core.SortingAnalyzer` handles the *sparsity* (see [Sparsity]_ section), i.e.,
+the channels on which waveforms and templates are defined on, for example, based on a physical distance from the
+channel with the largest peak amplitude.
+
+Most of the extensions live in the :code:`postprocessing` module, but there are some *core* extensions too:
+
+* select random spikes for downstream analysis (e.g., extracting waveforms or fitting a PCA model)
+* estimate templates
+* extract waveforms for single spikes
+* compute channel-wise noise levels
+
+.. note::
+
+    Some extensions depend on others, which must be pre-computed. For example :code:`waveforms`, the :code:`waveforms` extension depends on
+    the :code:`random_spikes` one. If the latter is not computed, computing :code:`waveforms` will throw an error.
+
+
+.. code-block:: python
+    from spikeinterface import create_sorting_analyzer
+
+    # create in-memory sorting analyzer object
+    sorting_analyzer = create_sorting_analyzer(
+        sorting=sorting,
+        recording=recording,
+        sparse=True, # default
+        format="memory", # default
+    )
+
+    print(sorting_analyzer)
+
+.. code-block:: bash
+
+    >>> SortingAnalyzer: 4 channels - 10 units - 1 segments - memory - sparse - has recording
+    >>> Loaded 0 extensions:
+
+The :code:`sorting_analyzer` object implements convenient functions to access the underlying :code:`recording` and
+:code:`sorting` objects' information:
+
+.. code-block:: python
+
+    num_channels = sorting_analyzer.get_num_channels()
+    num_units = sorting_analyzer.get_num_units()
+    sampling_frequency = sorting_analyzer.sampling_frequency
```

Comment:
```suggestion
    sampling_frequency = sorting_analyzer.sampling_frequency()
```

We could do this as the function instead of the attribute right, just to make everything be a function for the docs?

Reply:
we also have get_sampling_frequncy() but I think it is OK to show attributes also no ?

Reply:
It's just a pedagogical difference. For a beginner they have to learn both the library and the difference between a function and an attribute. So my style is more beginner-friendly in that we remove them having to learn a bunch of python features by saying look everything has a function to grab a piece of data. Right now if I read this I would ask myself my sampling_frequency is special and doesn't have a function whereas all the others have functions and that might distract me from the rest of the docs.

Doing a mix is better for the intermediate learning because they already know enough basics that they focus on the breadth of techniques. So they see that they can both grab the attribute or use a function and can chose their style. 

So for me it would be do we want this documentation to be more beginner or more intermediate friendly. My inclination is that people that read the docs need a bit more beginner friendly. So we give them one good solution and then once they improve their own python skills the'll realize pretty quickly that they can use the attribute if they want.

But this is a super small point.

Reply:
I am not convinced that functions are more beginner friendly than attributes.   Consistency I think is a good argument though.

Reply:
My argument was completely for consistency. I don't care if one does functions or attributes just be consistent and since most of them were already functions I chose the route of making all functions, but absolutely would be fine to just grab them all as attributes too (from my perspective).

Reply:
I normally use the attribute, which I think is easier. Maybe we can show both?

Reply:
Our argument (Heberto and mine since I think we fall in the same camp for documentation in general) is that if we include both we are making new learners/users learn both that Python has attributes + functions and the spikeinterface schema (why is it sampling_frequency and not samping_rate or why frequency and not period) so we are teaching them two things at once rather than teach them one thing (ie spikeinterface stuff). In our schema for docs you teach them one thing super well and then they can learn the other cool Python tricks themselves (function vs attribute). I think you and Sam like showing how many features the project has which is great for more experienced users/developers but I have a feeling that they don't read the docs (for example Aurélien once said he has never looked at the docs). 

My example would be as you learn English you could learn "need", "must", "require", which all have slightly different flavors so I could teach you that they all accomplish the same thing + teach you their different connotations or I could just teach you the word "need" and make sure you really understand that and even if there is a slightly better word "need" would still get the job done.

Reply:
Then in this case I would rather use `sorting_analyzer.sampling_frequency`, since this is the preferred use.

Reply:
Again it's a super small point and not a big deal for this PR :) Just something to keep in mind. We had a big discussion of some of this stuff here https://github.com/SpikeInterface/spikeinterface/issues/2327